target/linux/ramips/patches-6.6/860-ramips-add-eip93-driver.patch

```diff
@@ -2711,7 +2711,7 @@
 +	.remove_new = mtk_crypto_remove,
 +	.driver = {
 +		.name = "mtk-eip93",
-+		.of_match_table = of_match_ptr(mtk_crypto_of_match),
++		.of_match_table = mtk_crypto_of_match,
```

Comment:
We need it here. `of_match_ptr()` is used to address the scenario where the device table is not defined. When `CONFIG_OF is not set`, removing it will result in compilation errors.

Other files LGTM.

Reply:
there are non OF ramips targets?

Reply:
> there are non OF ramips targets?

All are OF based. But the device table is guarded by `if defined(CONFIG_OF)` here.
```
+#if defined(CONFIG_OF)
+static const struct of_device_id mtk_crypto_of_match[] = {
+	{ .compatible = "mediatek,mtk-eip93", },
+	{}
+};
+MODULE_DEVICE_TABLE(of, mtk_crypto_of_match);
+#endif
```
And this driver can also depend on COMPILE_TEST.
```
+config CRYPTO_DEV_EIP93
+	tristate "Support for EIP93 crypto HW accelerators"
+	depends on SOC_MT7621 || COMPILE_TEST
There is no need to touch this driver. It will be replaced by a new upstream one soon.
```
BTW. This driver will be replaced by a new upstream one soon.

Reply:
Ah yes I heard about this. Hopefully it ends up working. @Ansuel 's tests back in the day were not promising. I'll revert in the meantime.